app/src/test/java/com/jalasoft/routesapp/util/algorithm/RouteCalculatorTest.kt

```diff
@@ -0,0 +1,105 @@
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.HiltTestApplication
+import junit.framework.TestCase
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@ExperimentalCoroutinesApi
+@RunWith(RobolectricTestRunner::class)
+@Config(application = HiltTestApplication::class, manifest = Config.NONE)
+@HiltAndroidTest
+class RouteCalculatorTest : TestCase() {
+    @get:Rule
+    val hiltRule = HiltAndroidRule(this)
+
+    private lateinit var lines: List<LinePath>
+
+    @Before
+    public override fun setUp() {
+        super.setUp()
+        val start1 = RouteAlgorithmFakeData.coordinatesToLocation(-16.52035351419114, -68.12580890707301)
+        val end1 = RouteAlgorithmFakeData.coordinatesToLocation(-16.524285569842718, -68.12298370418992)
+        val points1 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.points1Array)
+        val stops1 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.stops1Array)
+        val line1 = LinePath("Line 1", "Bus", points1, start1, end1, stops1)
+
+        val start2 = RouteAlgorithmFakeData.coordinatesToLocation(-16.5255314, -68.1254204)
+        val end2 = RouteAlgorithmFakeData.coordinatesToLocation(-16.5241937, -68.1204527)
+        val points2 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.points2Array)
+        val stops2 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.stops2Array)
+        val line2 = LinePath("246", "Mini", points2, start2, end2, stops2)
+
+        val start3 = RouteAlgorithmFakeData.coordinatesToLocation(-16.52447, -68.12538)
+        val end3 = RouteAlgorithmFakeData.coordinatesToLocation(-16.5244779, -68.1253892)
+        val points3 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.points3Array)
+        val stops3 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.stops3Array)
+        val line3 = LinePath("LA", "Metro", points3, start3, end3, stops3)
+        hiltRule.inject()
+        lines = listOf(line1, line2, line3)
+    }
+
+    @Test
+    fun `Given a list of lines and an origin an destination point when the line stops are near the given points then it returns a list of single lines`() {
+        val originPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52094, -68.12419)
+        val destinationPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52355, -68.12269)
+        val minDistance = 200.0
+        val minDistanceBtwStops = 200.0
+
+        val result = RouteCalculator.calculate(lines, destinationPoint, originPoint, minDistance, minDistanceBtwStops)
+        assertTrue(result.isNotEmpty())
```

Comment:
Also here, check the result values, not only empty or not

Reply:
Added asserts for expected results